### PR TITLE
test: add tests for gitlab scrape function WIP

### PR DESCRIPTION
**Pull request by nsshaddox**
Not ready for review!

### CODE REVIEW

File: pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go [hunk 1]
```diff
@@ -152,3 +155,154 @@ func TestGetMergeRequests(t *testing.T) {
 		})
 	}
 }
+
+// func TestScrape(t *testing.T) {
```
Comment (densellp) [inline pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go:159]: You have this test commented out? is it still needed?

File: pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go [hunk 2]
```diff
@@ -152,3 +155,154 @@ func TestGetMergeRequests(t *testing.T) {
...
+// 				return
+// 			}
+// 		})
+// 	}
+// }
+
+func TestScrape(t *testing.T) {
+	type testCase struct {
+		name string
+		// bootTimeFunc        func(context.Context) (uint64, error)
```
Comment (densellp) [inline pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go:206]: are some of the comments in this test needed anymore?

File: pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go [hunk 3]
```diff
@@ -152,3 +155,154 @@ func TestGetMergeRequests(t *testing.T) {
...
+		// },
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			gls := newGitLabScraper(context.Background(), receivertest.NewNopCreateSettings(), &Config{MetricsBuilderConfig: tc.metricsConfig})
+			gls.cfg.HTTPClientSettings.Endpoint = tc.expectedEndpoint
+			gls.cfg.GitLabOrg = "liatrioinc"
+
+			// if tc.bootTimeFunc != nil {
```
Comment (densellp) [inline pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go:264]: Do we need these comments in this test anymore as well?

**Comment by adrielp** [comment]
There's conflicts here, bumping to see if this is still relevant?

**Pull request closed without merging**